src/UglyToad.PdfPig.Tokens/ObjectToken.cs

```diff
@@ -37,6 +37,39 @@ public ObjectToken(long position, IndirectReference number, IToken data)
             Data = data ?? throw new ArgumentNullException(nameof(data));
         }
 
+        /// <inheritdoc />
+        public bool Equals(IToken obj)
+        {
+            if (this == obj)
+                return true;
+
+            if (!(obj is ObjectToken other))
+            {
+                return false;
+            }
+
+            // I don't think we should compare this value...
+            // If this values are the same but this object are not the same (reference)
+            // It means that one of the obj is not from stream or one of the object have been overwritten
+            // So maybe putting an assert instead, would be a good idea?
+            if (Position != other.Position)
```

Comment:
This make sense? Maybe we should `throw` in this case

Reply:
I think maybe we should just remove the position check since you might have a situation where someone is accessing the document structure from different parsed documents using `document.Catalog.Structure` or however it's accessed and they share a producer which means they have identical objects but are in different positions and can be combined. The user can keep `ReferenceEquals` for the non-content aware equality.

Reply:
That make sense, I didn't think about the `ReferenceEquals`